platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/collections/AbstractLongList.java

```diff
@@ -243,8 +243,35 @@ protected AbstractLongList(
      * @param fileChannel the file channel to read the list body from
      * @throws IOException if there was a problem reading the file
      */
-    protected abstract void readBodyFromFileChannelOnInit(String sourceFileName, FileChannel fileChannel)
-            throws IOException;
+    protected void readBodyFromFileChannelOnInit(String sourceFileName, FileChannel fileChannel) throws IOException {
+        if (minValidIndex.get() < 0) {
+            // Empty list, nothing to read
+            return;
+        }
+
+        final int totalNumberOfChunks = calculateNumberOfChunks(size.get());
+        final int firstChunkWithDataIndex = toIntExact(minValidIndex.get() / numLongsPerChunk);
+        final int minValidIndexInChunk = toIntExact(minValidIndex.get() % numLongsPerChunk);
+
+        for (int chunkIndex = firstChunkWithDataIndex; chunkIndex < totalNumberOfChunks; chunkIndex++) {
+            final int startIndex = (chunkIndex == firstChunkWithDataIndex) ? minValidIndexInChunk : 0;
+            C chunk = readChunkData(fileChannel, chunkIndex, startIndex);
+            chunkList.set(chunkIndex, chunk);
```

Comment:
In case of `LongListDisk`, the method`readChunkData()` does two unrelated things: it reads a chunk and it writes the chunk to list's file. What about limiting `readChunkData()` to just reading a chunk? Then, introduce another method like `setChunk(index, chunk)`. This method will just `chunkList.set(index, chunk)`, and it will be overridden in `LongListDisk` to write the chunk to the temp file. What do you think about it?